src/compiler/emitter.ts

```diff
@@ -4836,7 +4842,7 @@ if (typeof __param !== "function") __param = function (paramIndex, decorator) {
                 var startIndex = emitDirectivePrologues(node.statements, /*startWithNewLine*/ false);
                 // Only Emit __extends function when target ES5.
                 // For target ES6 and above, we can emit classDeclaration as is.
-                if ((languageVersion < ScriptTarget.ES6) && (!extendsEmitted && resolver.getNodeCheckFlags(node) & NodeCheckFlags.EmitExtends)) {
+                if (!compilerOptions.noEmitHelpers && (languageVersion < ScriptTarget.ES6) && (!extendsEmitted && resolver.getNodeCheckFlags(node) & NodeCheckFlags.EmitExtends)) {

```

Comment:
`noEmitHelpers` should also exclude other helpers like `__decorate`, `__metadata`, and `__params`.